### ISSUE TITLE
Possible false positive for needless_lifetimes

### ISSUE THREAD

**Issue by paulkernfeld** [rust-lang-nursery/rust-clippy#2294]
Is it possible to omit the lifetimes from this expression? The [nomicon](https://doc.rust-lang.org/nomicon/lifetime-elision.html) doesn't provide any examples that cover this, as far as I can tell.

```rust
warning: explicit lifetimes given in parameter types where they could be elided
  --> src/properties.rs:22:5
   |
22 |     fn median<'a>(&'a self) -> Cow<'a, T>;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(needless_lifetimes)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.175/index.html#needless_lifetimes
```

I was able to accomplish this using the [`underscore_lifetimes` feature](https://github.com/rust-lang/rust/issues/44524), so maybe clippy should only emit a warning in this case if `underscore_lifetimes` is enabled?

I am using clippy 0.0.177.

**Comment by llogiq** [comment]
The second lifetime elision rule is: one lifetime for argument + returns can be elided. So what's wrong with the warning?

**Comment by paulkernfeld** [comment]
The problem was that I didn't understand how to omit the lifetime from the return type. The answer was that I needed to write this:

```rust
fn median(&self) -> Cow<T>;
```
